# Solution: Exploring Data

Hi,

The marketing team would like to share out the five happiest countries of the 2010s on social media.
I’ve attached a notebook that another data scientist started with happiness data inside. I would recommend:
* Creating a list of each country’s highest happiness score, and then sorting it from happiest to least happy country
* Creating a list of each country’s average happiness score, and then sorting it from happiest to least happy country

Are there any differences between the two lists?

Thanks!\
Anna

## Read in the data

In [1]:
# import the data 

import pandas as pd

df = pd.read_csv('../Data/happiness_survey_data.csv')
df.head()

country_name  year  happiness_score  social_support  \
0  Afghanistan  2008         3.723590        0.450662   
1  Afghanistan  2009         4.401778        0.552308   
2  Afghanistan  2010         4.758381        0.539075   
3  Afghanistan  2011         3.831719        0.521104   
4  Afghanistan  2012         3.782938        0.520637   

   freedom_to_make_life_choices  healthy_life_expectancy_at_birth  
0                      0.718114                         50.500000  
1                      0.678896                         50.799999  
2                      0.600127                         51.099998  
3                      0.495901                         51.400002  
4                      0.530935                         51.700001

In [2]:
# there are 2089 rows and 6 columns of data
df.shape

(2089, 6)

In [3]:
# the years range from 2005 - 2021
# happiness scores range from 2 - 8
df.describe()

year  happiness_score  social_support  \
count  2089.000000      2089.000000     2076.000000   
mean   2013.727621         5.473747        0.811542   
std       4.455614         1.115567        0.118935   
min    2005.000000         2.178809        0.290184   
25%    2010.000000         4.651972        0.747664   
50%    2014.000000         5.405246        0.834770   
75%    2017.000000         6.294282        0.904682   
max    2021.000000         8.018934        0.987343   

       freedom_to_make_life_choices  healthy_life_expectancy_at_birth  
count                   2057.000000                       2031.000000  
mean                       0.745462                         63.180326  
std                        0.140751                          6.948546  
min                        0.257534                          6.720000  
25%                        0.651689                         58.965000  
50%                        0.767357                         64.980003  
75%                        0.857677                         68.362499  
max                        0.985178                         74.349998

## Explore the data

In [4]:
# Filter out any data before 2010 and after 2019
filtered_df = df[(df.year >= 2010) & (df.year < 2020)]
filtered_df

country_name  year  happiness_score  social_support  \
2     Afghanistan  2010         4.758381        0.539075   
3     Afghanistan  2011         3.831719        0.521104   
4     Afghanistan  2012         3.782938        0.520637   
5     Afghanistan  2013         3.572100        0.483552   
6     Afghanistan  2014         3.130896        0.525568   
...           ...   ...              ...             ...   
2082     Zimbabwe  2015         3.703191        0.735800   
2083     Zimbabwe  2016         3.735400        0.768425   
2084     Zimbabwe  2017         3.638300        0.754147   
2085     Zimbabwe  2018         3.616480        0.775388   
2086     Zimbabwe  2019         2.693523        0.759162   

      freedom_to_make_life_choices  healthy_life_expectancy_at_birth  
2                         0.600127                         51.099998  
3                         0.495901                         51.400002  
4                         0.530935                         51.700001  
5                         0.577955                         52.000000  
6                         0.508514                         52.299999  
...                            ...                               ...  
2082                      0.667193                         51.200001  
2083                      0.732971                         51.674999  
2084                      0.752826                         52.150002  
2085                      0.762675                         52.625000  
2086                      0.631908                         53.099998  

[1412 rows x 6 columns]

In [5]:
# Group the data by country and calculate the maximum happiness score for each one
filtered_df.groupby('country_name')['happiness_score'].max()

country_name
Afghanistan    4.758381
Albania        5.867422
Algeria        6.354898
Angola         5.589001
Argentina      6.775805
                 ...   
Venezuela      7.478455
Vietnam        5.767344
Yemen          4.350313
Zambia         5.243996
Zimbabwe       4.955101
Name: happiness_score, Length: 164, dtype: float64

In [6]:
# Sort the grouped countries by happiness score and return the top five
filtered_df.groupby('country_name')['happiness_score'].max().sort_values(ascending=False).head()

country_name
Finland        7.858107
Denmark        7.788232
Switzerland    7.776209
Norway         7.678277
Canada         7.650346
Name: happiness_score, dtype: float64

In [7]:
# Group the data by country and calculate the average happiness score for each one
filtered_df.groupby('country_name')['happiness_score'].mean()

country_name
Afghanistan    3.501017
Albania        4.976791
Algeria        5.389717
Angola         4.420299
Argentina      6.398067
                 ...   
Venezuela      5.858085
Vietnam        5.278218
Yemen          3.765910
Zambia         4.452677
Zimbabwe       4.074384
Name: happiness_score, Length: 164, dtype: float64

In [8]:
# Sort the grouped countries by happiness score and return the top five
filtered_df.groupby('country_name')['happiness_score'].mean().sort_values(ascending=False).head()

country_name
Denmark        7.618252
Switzerland    7.568010
Finland        7.553138
Norway         7.541094
Iceland        7.518146
Name: happiness_score, dtype: float64

## Compare the two lists

### Happiest countries for social media
    
**Countries with the highest happiness scores**
1. Finland: 7.858107
2. Denmark: 7.788232
3. Switzerland: 7.776209
4. Norway: 7.678277
5. Canada: 7.650346

**Countries with the highest average happiness scores**
1. Denmark: 7.618252
2. Switzerland: 7.568010
3. Finland: 7.553138
4. Norway: 7.541094
5. Iceland: 7.518146

In [9]:
# Look at top few countries in more detail
filtered_df[filtered_df.country_name.isin(['Finland', 'Denmark', 'Switzerland'])]

country_name  year  happiness_score  social_support  \
481       Denmark  2010         7.770515        0.974977   
482       Denmark  2011         7.788232        0.961736   
483       Denmark  2012         7.519909        0.951437   
484       Denmark  2013         7.588607        0.964708   
485       Denmark  2014         7.507559        0.956344   
486       Denmark  2015         7.514425        0.959701   
487       Denmark  2016         7.557783        0.954452   
488       Denmark  2017         7.593702        0.952100   
489       Denmark  2018         7.648786        0.958219   
490       Denmark  2019         7.693003        0.957706   
590       Finland  2010         7.393264        0.935481   
591       Finland  2011         7.354225        0.937857   
592       Finland  2012         7.420209        0.927739   
593       Finland  2013         7.444636        0.940869   
594       Finland  2014         7.384571        0.952017   
595       Finland  2015         7.447926        0.947801   
596       Finland  2016         7.659843        0.953940   
597       Finland  2017         7.788252        0.963826   
598       Finland  2018         7.858107        0.962155   
599       Finland  2019         7.780348        0.937416   
1773  Switzerland  2012         7.776209        0.946864   
1774  Switzerland  2014         7.492804        0.958796   
1775  Switzerland  2015         7.572137        0.938334   
1776  Switzerland  2016         7.458520        0.927628   
1777  Switzerland  2017         7.473593        0.949661   
1778  Switzerland  2018         7.508587        0.930291   
1779  Switzerland  2019         7.694221        0.948513   

      freedom_to_make_life_choices  healthy_life_expectancy_at_birth  
481                       0.943631                         69.400002  
482                       0.934760                         69.620003  
483                       0.932628                         69.839996  
484                       0.920255                         70.059998  
485                       0.941572                         70.279999  
486                       0.941436                         70.500000  
487                       0.948231                         70.625000  
488                       0.955416                         70.750000  
489                       0.935438                         70.875000  
490                       0.963318                         71.000000  
590                       0.916009                         69.599998  
591                       0.936448                         69.820000  
592                       0.920968                         70.040001  
593                       0.918625                         70.260002  
594                       0.933044                         70.480003  
595                       0.929862                         70.699997  
596                       0.948372                         70.775002  
597                       0.962199                         70.849998  
598                       0.937807                         70.925003  
599                       0.947617                         71.000000  
1773                      0.945428                         71.519997  
1774                      0.949401                         71.639999  
1775                      0.927802                         71.699997  
1776                      0.933947                         71.900002  
1777                      0.924997                         72.099998  
1778                      0.926415                         72.300003  
1779                      0.913167                         72.500000